drivers/timer/cortex_m_systick.c

```diff
@@ -421,13 +480,16 @@ void sys_clock_idle_exit(void)
 #endif /* CONFIG_CORTEX_M_SYSTICK_IDLE_TIMER */
 
 	if (last_load == TIMER_STOPPED) {
-		SysTick->CTRL |= SysTick_CTRL_ENABLE_Msk;
+		sys_write32(SYST_CSR_RUN, SYST_CSR);
 	}
 }
 
 void sys_clock_disable(void)
 {
-	SysTick->CTRL &= ~SysTick_CTRL_ENABLE_Msk;
+	K_SPINLOCK(&lock) {
+		sys_write32(0, SYST_CSR);
```

Comment:
Unlike the original code, this is also changing the clock source and the interrupt enable.
Any reason for that change?

Reply:
These bits are re-enabled by the SYST_CSR_RUN macro later, but the missing read op here for TICKINT clearing is a bug.

Reply:
Alternate clock source isn’t used in Zephyr yet but there is an issue requesting its addition.

Reply:
> Alternate clock source isn’t used in Zephyr yet but there is an issue requesting its addition.

It was my issue and frozen PR :). This PR is a cutted down version of the previous one.

Reply:
I mean, is it some delay will be add if CLKSOURCE will be changed in the disabled state of the SysTick ? If not, then using 0 for the inactive CSR would be reasonable, and nasty TIMER_STOPPED can be thrown away. ARM's  public docos tell nothing about this.

BTW I've remember another reason to move away from the current CMSIS based implementation: multicore. In the current implementation, the base address is hardcoded, but it can be wrong in multicore case.

Reply:
> In the current implementation, the base address is hardcoded, but it can be wrong in multicore case.

It is not wrong. The SYSTick lives on the Private Peripheral Bus that is core specific.
That is to say the SysTick that core0 accesses is a different one that of what core1’s accessing.

Reply:
Ok, I've fixed the 'last_loaded' related bug. Could you recheck the latest version?